rbx_binary/src/deserializer.rs

```diff
@@ -263,6 +264,8 @@ struct BinaryDeserializer<R> {
     /// are interpreted by Roblox.
     metadata: HashMap<String, String>,
 
+    shared_strings: HashMap<u32, SharedString>,
```

Comment:
This does not need to be a `HashMap`, we should be able to get away with this being a `Vec` instead. In general, any time we have a map of contiguous integer keys, we can use an array-like data structure instead of a map-like one.

Can we also add a comment like the other members of this struct describing what's in here? The key term here is probably "shared string dictionary".